pydra/engine/submitter.py

```diff
@@ -123,19 +124,19 @@ def __exit__(self, type, value, traceback):
                     f'Submitting runnable {job}{str(sidx) if sidx is not None else ""}'
                 )
                 if is_workflow(runnable):
-                    futures.add(asyncio.create_task(job.run(self)))
+                    # job has no state anymore
+                    futures.add(asyncio.create_task(job._run(self)))
                 else:
                     # tasks are submitted to worker for execution
                     futures.add(self.worker.run_el(job))
         else:
-            job = runnable.to_job(None)
-            job.results_dict[None] = (None, job.checksum)
+            runnable.results_dict[None] = (None, runnable.checksum)
             if is_workflow(runnable):
                 # this should only be reached through the job's `run()` method
-                runnable = await self._run_workflow(job)
+                runnable = await self._run_workflow(runnable)
```

Comment:
now that we aren't copying `runnable` to a job, are these returns/assignments necessary? I think we could get away with just a 
```suggestion
               await self._run_workflow(runnable)
```
here